prime-router/src/test/kotlin/fhirengine/engine/FhirReceiverFilterTests.kt

```diff
@@ -0,0 +1,2159 @@
+        mappedConditionFilter: ReportStreamConditionFilter = emptyList(),
+    ) {
+        every { getJurisFilters(any(), any()) } returns jurisFilter
+        every { getQualityFilters(any(), any()) } returns qualFilter
+        every { getRoutingFilter(any(), any()) } returns routingFilter
+        every { getProcessingModeFilter(any(), any()) } returns processingModeFilter
+        every { getConditionFilter(any(), any()) } returns conditionFilter
+        every { getMappedConditionFilter(any(), any()) } returns mappedConditionFilter
+    }
+
+    @BeforeEach
+    fun reset() {
+        actionHistory.reportsIn.clear()
+        actionHistory.reportsOut.clear()
+        actionHistory.actionLogs.clear()
+        azureEventService.clear()
+        clearAllMocks()
+    }
+
+    @Test
+    fun `test applyFilters receiver setting - (reverseTheQualityFilter = true) `() {
+        val fhirData = File(QUALITY_TEST_URL).readText()
+        val bundle = FhirTranscoder.decode(fhirData)
+        val settings = FileSettings().loadOrganizations(orgWithReversedQualityFilter)
+        val jurisFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val qualFilter = listOf(PROVENANCE_COUNT_EQUAL_TO_TEN)
+        val routingFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val processingModeFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val conditionFilter = emptyList<String>()
+        val mappedConditionFilter = emptyList<ConditionFilter>()
+
+        val engine = spyk(makeFhirEngine(metadata, settings) as FHIRRouter)
+        engine.setFiltersOnEngine(
+            jurisFilter, qualFilter, routingFilter, processingModeFilter, conditionFilter, mappedConditionFilter
+        )
+
+        // act
+        val receivers = engine.findReceiversForBundle(bundle, report.id, actionHistory, Topic.FULL_ELR)
+
+        // assert
+        assertThat(receivers).isNotEmpty()
+    }
+
+    // TODO: review
+    @Test
+    fun `test applyFilters receiver setting - (reverseTheQualityFilter = false) `() {
+        val fhirData = File(QUALITY_TEST_URL).readText()
+        val bundle = FhirTranscoder.decode(fhirData)
+        val settings = FileSettings().loadOrganizations(oneOrganization)
+        val jurisFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val qualFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val routingFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val processingModeFilter = listOf(PROVENANCE_COUNT_GREATER_THAN_ZERO)
+        val conditionFilter = emptyList<String>()
+        val mappedConditionFilter = emptyList<ConditionFilter>()
+
+        val engine = spyk(makeFhirEngine(metadata, settings) as FHIRRouter)
+        engine.setFiltersOnEngine(
+            jurisFilter, qualFilter, routingFilter, processingModeFilter, conditionFilter, mappedConditionFilter
+        )
+
+        // act
+        val receivers = engine.findReceiversForBundle(bundle, report.id, actionHistory, Topic.FULL_ELR)
```

Comment:
Should be calling FhirReciverFilter.kt instead of FHIRRouter.kt but maybe this is part of the todos.

Reply:
Addressed. This got stuck shelved in intellij.